.github/workflows/update-insecure-dependencies.yaml

```diff
@@ -55,7 +53,22 @@ jobs:
       - name: "Update dependencies"
         id: update
         run: |
-          osv-scanner --lockfile=go.mod --json | jq '.results[].packages[].package.name' | grep -Ev ${{ env.IGNORE_PACKAGES }} | xargs -I {} go get -u {}
+          for dep in $(osv-scanner --lockfile=go.mod --json | jq -c '.results[].packages[] | .package.name as $vulnerablePackage | {
+            name: $vulnerablePackage,
+            current: .package.version,
+            fixedVersions: [.vulnerabilities[].affected[] | select(.package.name == $vulnerablePackage) | .ranges[].events[] | select(.fixed != null) | .fixed] | unique
+          } | select(.fixedVersions | length > 0)'); do
+            IFS=. read -r currentMajor currentMinor currentPatch <<< "$(jq -r .current <<< "$dep")"
+            # Update to the first version that's greater than our current version
+            for version in $(jq -cr .fixedVersions[] <<< "$dep" | sort -V); do # sort supports semver sort
+              IFS=. read -r fixMajor fixMinor fixPatch <<< "$version"
+              if ((currentMajor <= fixMajor)) && ((currentMinor <= fixMinor)) && ((currentPatch <= fixPatch)); then
```

Comment:
so this is so we don't downgrade to a version that is not vulnerable right?

Reply:
yep